java/code/src/com/suse/manager/webui/services/pillar/MinionGeneralPillarGenerator.java

```diff
@@ -56,8 +57,12 @@
      * @return the SaltPillar containing the pillar data
      */
     @Override
-    public Optional<SaltPillar> generatePillarData(MinionServer minion) {
-        SaltPillar pillar = new SaltPillar();
+    public Optional<Pillar> generatePillarData(MinionServer minion) {
+        Pillar pillar = minion.getPillarByCategory(CATEGORY).orElseGet(() -> {
+            Pillar newPillar = new Pillar(CATEGORY, new HashMap<>(), minion);
+            minion.getPillars().add(newPillar);
+            return newPillar;
+        });
```

Comment:
Maybe it should call `pillar.getPillar().clear()` here, to be sure that there are no obsolete entries left.

The same in other Pillar Generators.

Reply:
Isn't this implied by being in `orElseGet` part?

Reply:
well, either we find the pillar of the category we need to generate and reuse it (no need of cleaning) or e got a newly generated one (and added to the collection) from `orElseGet()`. When reusing we may have existing entries to be cleared: you are right!

Reply:
fixed